src/common/infra/storage/mod.rs

```diff
@@ -48,7 +48,14 @@ pub async fn list(prefix: &str) -> Result<Vec<String>, anyhow::Error> {
     let files = DEFAULT
         .list(Some(&prefix.into()))
         .await?
-        .map_ok(|meta| meta.location.to_string())
+        .map_ok(|meta| {
+            let file = meta.location.to_string();
+            let columns = file.split('/').collect::<Vec<&str>>();
+            metrics::STORAGE_READ_REQUESTS
+                .with_label_values(&[columns[1], columns[3], columns[2]])
+                .inc();
+            file
+        })
```

Comment:
I suggest removing this section, the `list` operation is only one request, but here you add a request for each file. if the `list` returns thousands of files, it will increase thousands of requests.